products/bash/configure-postgres.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+#******************************************************************************
+# Licensed Materials - Property of IBM
+# (c) Copyright IBM Corporation 2019. All Rights Reserved.
+#
+# Note to U.S. Government Users Restricted Rights:
+# Use, duplication or disclosure restricted by GSA ADP Schedule
+# Contract with IBM Corp.
+#******************************************************************************
+
+function usage {
+  echo "Usage: $0 -n <namespace>"
+}
+
+namespace="cp4i"
+
+while getopts "n:r:" opt; do
+  case ${opt} in
+    n ) namespace="$OPTARG"
+      ;;
+    \? ) usage; exit
+      ;;
+  esac
+done
+
+namespace=$(echo $namespace | sed 's/-/_/g')
+
+if ! oc exec -n postgres -it $(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}') \
+-- psql -U ${namespace} -d db_${namespace} -c '\l' ; then
+    echo "INFO: Creating Database db_${namespace} , User ${namespace}, "
+    oc exec -n postgres -it $(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}') \
+    -- psql  << EOF
+CREATE DATABASE db_${namespace};
+CREATE USER ${namespace} WITH PASSWORD 'password';
+GRANT CONNECT ON DATABASE db_${namespace} TO ${namespace};
+EOF
+else
+  echo "INFO: Table already exists, skipping this step"
+fi
+
+if ! oc exec -n postgres -it $(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}') \
+-- psql -U ${namespace} -d db_${namespace} -c '\dt'  | grep -i quotes ; then
+
+echo "INFO: Creating tables in the database db_${namespace}"
+  oc exec -n postgres -it $(oc get pod -n postgres -l name=postgresql -o jsonpath='{.items[].metadata.name}') \
```

Comment:
I think only one oc exec should be used, the 2nd one, with the if checking the result of the 2nd one

Reply:
Removed the second if check and added IF NOT EXISTS